### PR TITLE
[DEV APPROVED] Add redirect feature to fincap site

### PR DESCRIPTION
**Pull request by tomas-stefano**
[TP card](https://moneyadviceservice.tpondemand.com/entity/9225-implement-internal-redirect-feature-on-fincap)

## Solution

The redirect mechanism is implemented as follows:

- The user visits a page that does not exist:

  Fincap app will request to CMS if the page contains a redirect. If it does, it will redirect the user; if not normal Rails flow will follow.

- The user visits any supported page type but the page does not exist:

  If there is a redirect on CMS, Fincap app will catch the redirect exception from mas-cms-client gem
  and redirect the user to the destination specified on CMS.

In all other cases, the app will continue to run and will return the 'not found' page. 

## Observations

This depends on this PR https://github.com/moneyadviceservice/cms/pull/480
Please review this PR as well.

### CODE REVIEW

**Comment by tomas-stefano** [comment]
@margOnline I added one. Can you check please?

File: features/redirect.feature [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  Scenario: Visiting an old site vanity url
+    Given I visit the page '/young-adults'
+    Then I should be reading the page '/en/lifestages/young-adults'
+
+  Scenario: Visiting an old site news page
+    Given I visit the page '/en/news/old-news'
+    Then I should be reading the page '/en/news/press-release-a-new-way-to-pay'
+
+  @allow-rescue
+  Scenario: Visiting a page that do not exist and does not have a redirect
```
Comment (margOnline) [inline features/redirect.feature]: `Visiting a page that does not exist and does not have a redirect`

File: features/redirect.feature [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    Given I visit the page '/this-page-dont-exist-anywhere'
+    Then I should be reading the page not found
```
Comment (margOnline) [inline features/redirect.feature]: `Then I should see the page not found` wdyt?

Comment (tomas-stefano) [reply features/redirect.feature]: Done ♻️

File: features/redirect.feature (hunk 1 above)
Comment (tomas-stefano) [reply features/redirect.feature]: Done ♻️

**Comment by margOnline** [comment]
nice work

**Pull request merged**